packages/nitro-protocol/docs/adjudicator/state-format.md

```diff
@@ -73,13 +73,13 @@ which contains fields which are allowed to change. These structs, along with rem
 
 ## Implementation FAQs
 
-**Why include the ChannelId separately?** We have to calculate the `channelId` for every single operation anyway. Given that we have it it's cheaper to hash in the hash, rather than the individual components again.
+**Why include the `channelId` in the State, rather than its pre-image?** It is necessary for the chain to calculate (and hence store in memory) the `channelId` for every state-changing method, in order to read the correct slot from storage. When computing the hash of a State struct (which often happens next), it is more efficient (in terms of gas consumption) to reuse `channelId`, rather than to (re)hash the pre-image of `channelId` along with the rest of the data. This is because `channelId` is generally shorter than its pre-image, and the gas costs of hashing increase with the length of the input.
```

Comment:
What do you mean by `pre-image` exactly? Is it the state from before? Maybe it is a term that I just don't know?
Other than that this looks much more readable :)